packages/material-ui/src/Dialog/Dialog.js

```diff
@@ -279,12 +283,12 @@ Dialog.propTypes = {
   /**
    * Dialog children, usually the included sub-components.
    */
-  children: PropTypes.node.isRequired,
+  children: PropTypes.node,
```

Comment:
Not sure what the point is making a `node` required. One could still omit content by passing `false`. TypeScript allowed `undefined | null` since the beginning.

Reply:
It was meant to tell users that instead of
```jsx
<Dialog>{condition ? <B /> : null}</Dialog>
```

they might prefer

```jsx
{condition ? <Dialog><B /></Dialog> : null}
```

But no strong preference

Reply:
> `<Dialog>{condition ? <B /> : null}</Dialog>`

This is how I prefer to write conditional rendering as well but I see very many people defaulting to shorthand `<Dialog>{condition && <B />}</Dialog>` which does pass `PropTypes.node.isRequired`

I wouldn't mind adding a  warning with a rationale why one pattern should be preferred but then it needs a rationale, an actionable hint and needs to be applied consistently.